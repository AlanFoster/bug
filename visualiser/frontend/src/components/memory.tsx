import * as React from "react";
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles(theme => ({
    root: {
    },
    memoryDump: {
    }
}));

type Props = {
    readonly memory: ArrayBuffer;
}

export const Memory: React.FC<Props> = function ({ memory }) {
    const classes = useStyles({});
    const int8Buffer = new Int8Array(memory);
    const memoryWidth = int8Buffer.length.toString().length;
    let memoryDump = "";
    for (let i = 0; i < int8Buffer.length; i += 4) {
        memoryDump += i.toString().padStart(memoryWidth, '0') + ' | ';
        memoryDump += [int8Buffer[i], int8Buffer[i + 1], int8Buffer[i + 2], int8Buffer[i + 3]].join(' ');
        memoryDump += "\n";
    }

    return (
        <div className={classes.root}>
            <div>Memory</div>
            <pre className={classes.memoryDump}>{memoryDump}</pre>
        </div>
    )
}
