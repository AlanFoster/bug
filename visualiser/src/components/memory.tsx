import * as React from "react";

export const Memory = function () {
    return (
        <React.Fragment>
            Memory
        </React.Fragment>
    )
}
