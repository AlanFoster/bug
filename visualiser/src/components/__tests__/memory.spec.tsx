import * as React from "react";
import renderer from "react-test-renderer"
import { Memory } from '../memory';

describe("Memory", function () {
    it("renders successfully", function () {
        const tree = renderer.create(<Memory/>);

        expect(tree).toMatchSnapshot();
    });
});
