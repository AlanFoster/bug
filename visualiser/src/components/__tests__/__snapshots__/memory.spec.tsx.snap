// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Memory renders successfully 1`] = `"Memory"`;
